crates/services/gas_price_service/src/v1/metadata.rs

```diff
@@ -3,7 +3,10 @@ use fuel_gas_price_algorithm::v1::{
     AlgorithmUpdaterV1,
     L2ActivityTracker,
 };
-use std::num::NonZeroU64;
+use std::{
+    collections::BTreeMap,
+    num::NonZeroU64,
+};
 
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq)]
 pub struct V1Metadata {
```

Comment:
Do we have a task to remove `unrecorded_blocks` field from the `V1Metadata`?(We should use separate table to store them). Taking into account that committer posts bundles each hour, it can be a lot of storage space(each `V1Metadata` will have 3600 entries, and we will insert it into the storage for each block height).

Reply:
That's a good idea. I'll add a story. It doesn't need to be removed from `V1Metadata`, it just needs to be handled in the adapter.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2454

Reply:
I prefer the usage of the `unrecorded_blocks_bytes` field over fetching all unrecorded blocks on startup-up from the storage. Iterating over all elements of the storage only works on the `latest_view`. If you want to initiate the gas price service at a specific block height via `view_at,` then fetching will not work.

We don't have use cases right now where we need to support `view_at`, so it is not critical. But it would be nice if we had this option.

Reply:
Is this more about the next comment:

> Why unrecorded_blocks_bytes can't be a field of the metadata, instead of calculating it each time?

We still need the bytes for each unrecorded block, even if we store the total as well.

Reply:
> We still need the bytes for each unrecorded block, even if we store the total as well.

Yes, we need it, but we can fetch this information during iteration over `heights` requesting `maybe_bytes` from the table in the database.

<img width="516" alt="image" src="https://github.com/user-attachments/assets/c590cf4e-a218-4c8e-9a6e-6566e907a24c">